plugin/azure/setup_test.go

```diff
@@ -0,0 +1,72 @@
+package azure
+
+import (
+	"testing"
+
+	"github.com/caddyserver/caddy"
+)
+
+func TestSetup(t *testing.T) {
+	tests := []struct {
+		body          string
+		expectedError bool
+	}{
+		{`azure`, false},
+		{`azure :`, true},
+		{`azure resource-set:hosted-zone`, false},
+		{`azure resource-set:hosted-zone {
+    tenant_id
+}`, true},
+		{`azure resource-set:hosted-zone {
+    tenant_id
+}`, true},
+		{`azure resource-set:hosted-zone {
+    client_id
+}`, true},
+		{`azure resource-set:hosted-zone {
+    client_secret
+}`, true},
+		{`azure resource-set:hosted-zone {
+    subscription_id
+}`, true},
+		{`azure resource-set:hosted-zone {
+    upstream 10.0.0.1
+}`, true},
+
+		{`azure resource-set:hosted-zone {
+    upstream
+}`, true},
+		{`azure resource-set:hosted-zone {
+    foobar
+}`, true},
+		{`azure resource-set:hosted-zone {
+    tenant_id <tenant_id>
```

Comment:
don't bring these <foo> in back in the tests. This also still needs to be fixed in the README.md itself.